### PR TITLE
update doc for Adapter and Device

### PR DESCRIPTION
**Pull request by Michael-Lfx**
Fixes #issue
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:
- [x] `rustfmt` run on changed code

### CODE REVIEW

File: src/hal/src/device.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 //! and managing graphics resources such as buffers, images and memory.
 //!
 //! The `Adapter` and `Device` types are very similar to the Vulkan concept of
-//! "physical devices" vs. "logical devices"; an `Adapter` is single GPU
+//! "logical devices" vs. "physical devices"; an `Adapter` is single GPU
```
Comment (kvark) [inline src/hal/src/device.rs]: I don't think this change is correct. The line above says "Adapter" followed by "Device, so here it should be "physical device" followed by "logical device". The old comment is correct, therefore.

Comment (Michael-Lfx) [reply src/hal/src/device.rs]: Here’s to say the “Adapter” equals to “physical device” and “Device” is “logical device”?

Comment (kvark) [reply src/hal/src/device.rs]: That is correct, yes

Comment (Michael-Lfx) [reply src/hal/src/device.rs]: Thank you.

Comment (grovesNL) [reply src/hal/src/device.rs]: @kvark what do you think about aligning the naming with Vulkan here?

**Comment by kvark** [comment]
We do have a PhysicalDevice, it's just included in the Adapter with more info, so I don't think getting closer to Vk would be giving us much.

**Comment by grovesNL** [comment]
Yeah I was just wondering because I think a couple others were confused about this too (can't search for "adapter" in Vulkan docs etc.)

**Pull request closed without merging**